zulipterminal/model.py

```diff
@@ -1224,7 +1225,13 @@ def make_reduced_stream_data(stream: Subscription) -> StreamData:
             # different formats
             subscription["color"] = canonicalize_color(subscription["color"])
 
-            self.stream_dict[subscription["stream_id"]] = subscription
+            stream_id = subscription["stream_id"]
+            if stream_id in self.unsubscribed_stream_dict:
+                del self.unsubscribed_stream_dict[stream_id]
+            elif stream_id in self.never_subscribed_stream_dict:
+                del self.never_subscribed_stream_dict[stream_id]
+            self.stream_dict[stream_id] = subscription
```

Comment:
I understand that this is aimed at ensuring that the stream 'dicts' are synchronized, and corresponds to the code in the other function, but it seems strange included within the subscription-removal commit, since nothing is calling them except the initial call to this function, when the synchronization *should* be a no-op.

It may be clearer to consider adding and removing subscriptions separately, commit-wise?